src/main/scala/firrtl/Emitter.scala

```diff
@@ -290,89 +363,124 @@ class VerilogEmitter extends SeqTransform with Emitter {
       case (i: Info) =>
         i match {
-          case NoInfo => // Do nothing
+          case NoInfo => colNum // Do nothing
           case f: FileInfo =>
             val escaped = FileInfo.escapedToVerilog(f.escaped)
             w.write(s" // @[$escaped]")
+            colNum
           case m: MultiInfo =>
             val escaped = FileInfo.escapedToVerilog(m.flatten.map(_.escaped).mkString(" "))
             w.write(s" // @[$escaped]")
+            colNum
         }
       case (s: Seq[Any]) =>
-        s.foreach(emit(_, top + 1))
-        if (top == 0) w.write("\n")
+        val nextColNum = s.foldLeft(colNum) {
+          case (colNum, e) => emitCol(e, top + 1, tabs, colNum)
+        }
+        if (top == 0) {
+          w.write("\n")
+          0
+        } else {
+          nextColNum
+        }
       case x => throwInternalError(s"trying to emit unsupported operator: $x")
     }
   }
 
   //;------------- PASS -----------------
-  def v_print(e: Expression)(implicit w: Writer) = e match {
+  def v_print(e: Expression, colNum: Int)(implicit w: Writer) = e match {
     case UIntLiteral(value, IntWidth(width)) =>
-      w.write(s"$width'h${value.toString(16)}")
+      val contents = s"$width'h${value.toString(16)}"
+      w.write(contents)
+      colNum + contents.size
     case SIntLiteral(value, IntWidth(width)) =>
       val stringLiteral = value.toString(16)
-      w.write(stringLiteral.head match {
+      val contents = stringLiteral.head match {
         case '-' if value == FixAddingNegativeLiterals.minNegValue(width) => s"$width'sh${stringLiteral.tail}"
         case '-'                                                          => s"-$width'sh${stringLiteral.tail}"
         case _                                                            => s"$width'sh${stringLiteral}"
-      })
+      }
+      w.write(contents)
+      colNum + contents.size
     case _ => throwInternalError(s"attempt to print unrecognized expression: $e")
   }
 
   // NOTE: We emit SInts as regular Verilog unsigned wires/regs so the real type of any SInt
   // reference is actually unsigned in the emitted Verilog. Thus we must cast refs as necessary
   // to ensure Verilog operations are signed.
   def op_stream(doprim: DoPrim): Seq[Any] = {
+    def parenthesize(e: Expression, isFirst: Boolean): Any = doprim.op match {
+      case Shl | Cat | Cvt | AsUInt | AsSInt | AsClock | AsAsyncReset => e
+      case _ =>
+        e match {
+          case e: DoPrim =>
+            op_stream(e) match {
+              case Seq(passthrough: Expression) => parenthesize(passthrough, isFirst)
+              case other =>
+                if (
+                  VerilogEmitter
+                    .precedenceGt(e.op, doprim.op) || (VerilogEmitter.precedenceGeq(e.op, doprim.op) && isFirst)
+                ) {
+                  other
+                } else {
+                  Seq("(", other, ")")
+                }
+            }
```

Comment:
Import `VerilogEmitter._` above, then we can make stuff fit on single lines. The ScalaFmt should support putting stuff on the same line (like the `case e: DoPrim => op_stream(e)`), but I don't think it'll go backwards for you. I could also be wrong
```suggestion
          case e: DoPrim => op_stream(e) match {
            case Seq(passthrough: Expression) => parenthesize(passthrough, isFirst)
            case other =>
              if (precedenceGt(e.op, doprim.op) || (precedenceGeq(e.op, doprim.op) && isFirst)) {
                other
              } else {
                Seq("(", other, ")")
              }
          }
```
I also find this logic pretty hard to understand, a couple of comments about how `parenthesize` works would be helpful.